### PR TITLE
DROOLS-5960 : GDST V&V should support Dates

### PR DESCRIPTION
**Pull request by Rikkola**
**JIRA**: 

[DROOLS-5960](https://issues.redhat.com/browse/DROOLS-5960)

**referenced Pull Requests**: 

* https://github.com/kiegroup/drools/pull/3333
* https://github.com/kiegroup/drools-wb/pull/1457

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by Rikkola** [comment]
@jomarko Hi, setting this PR for reviews, but no hurry to do it in this sprint.

**Review by vpellegrino — APPROVED** [review]
What I see here, for the Java part, looks good to me. Unfortunately, I am missing some basic information about `.gdst` XML files. But generally speaking, thanks for this PR @Rikkola

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/test/java/org/drools/workbench/services/verifier/plugin/client/fromfile/DatesFromFileTest.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.drools.workbench.services.verifier.plugin.client.testutil.TestUtil.assertContains;
+import static org.drools.workbench.services.verifier.plugin.client.testutil.TestUtil.assertDoesNotContain;
+import static org.drools.workbench.services.verifier.plugin.client.testutil.TestUtil.loadResource;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DatesFromFileTest extends AnalyzerUpdateTestBase {
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/test/java/org/drools/workbench/services/verifier/plugin/client/fromfile/DatesFromFileTest.java:34]: Thank you for this test @Rikkola , I tried to open GDST in decision central, I can confirm validation works but reports missing range in inappropriate order.

See this PR https://github.com/kiegroup/kie-wb-common/pull/3569

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/test/java/org/drools/workbench/services/verifier/plugin/client/fromfile/DatesFromFileTest.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        final GuidedDecisionTable52 table52 = GuidedDTXMLPersistence.getInstance().unmarshal(xml);
+
+        final Analyzer analyzer = analyzerProvider.makeAnalyser(table52);
+
+        analyzer.resetChecks();
+        analyzer.analyze();
+
+        assertContains(analyzerProvider.getAnalysisReport(),
+                       CheckType.MISSING_RANGE,
+                       Severity.NOTE);
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/test/java/org/drools/workbench/services/verifier/plugin/client/fromfile/DatesFromFileTest.java:51]: Please update:
```
assertContains(analyzerProvider.getAnalysisReport(),
                       CheckType.MISSING_RANGE,
                       Severity.NOTE,
                       1, 2, 3);
```

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/test/java/org/drools/workbench/services/verifier/plugin/client/fromfile/DatesFromFileTest.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/test/java/org/drools/workbench/services/verifier/plugin/client/fromfile/DatesFromFileTest.java:69]: Please update:
```
assertContains(analyzerProvider.getAnalysisReport(),
                       CheckType.MISSING_RANGE,
                       Severity.NOTE,
                       1, 2, 3);
```

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/test/java/org/drools/workbench/services/verifier/plugin/client/fromfile/DatesFromFileTest.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/test/java/org/drools/workbench/services/verifier/plugin/client/fromfile/DatesFromFileTest.java:104]: Please update
```
assertContains(analyzerProvider.getAnalysisReport(),
                       CheckType.MISSING_RANGE,
                       Severity.NOTE,
                       1, 2);
```

**Review by jomarko — CHANGES_REQUESTED** [review]
Thank you for the contribution @Rikkola . I really like the coverage we add. However few comments before merge.

**Comment by Rikkola** [comment]
@jomarko FYI I am investigating the issues you found.
@vpellegrino @jomarko I also started to prepare a PR that moves the "integration tests" to drools. Probably better to have them there, but I would merge that as a separate PR and target 7.11 with it.

**Comment by jomarko** [comment]
@Rikkola not sure what investigation you mean. I just require few updates to tests
- https://github.com/kiegroup/drools-wb/pull/1457#pullrequestreview-592134612
- https://github.com/Rikkola/drools/pull/9/files

Sorting in validation report should be fixed by:
- https://github.com/kiegroup/kie-wb-common/pull/3569

**Comment by Rikkola** [comment]
@jomarko Ok. I think I got all the bugs and improved the catches so that the error does not jump on your face, but is more detailed in the V&V reporting panel.

**Comment by jomarko** [comment]
@Rikkola we have a test failing `org.drools.workbench.services.verifier.plugin.client.fromfile.DatesFromFileTest.testGap`, seems be related to your changes.

**Review by jomarko — CHANGES_REQUESTED** [review]
https://github.com/kiegroup/drools-wb/pull/1457#issuecomment-789505545

**Comment by Rikkola** [comment]
@jomarko Checking

**Comment by Rikkola** [comment]
@jomarko Ok so the gap test had just 2 lines, fixed it :)

**Comment by jomarko** [comment]
Jenkins do fdb

**Comment by Rikkola** [comment]
`org.kie.bc.DeploymentIT.org.kie.bc.DeploymentIT`

Jenkins do fdb

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/test/java/org/drools/workbench/services/verifier/plugin/client/fromfile/DatesFromFileTest.java (hunk 1 above)
Comment (Rikkola) [reply drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/test/java/org/drools/workbench/services/verifier/plugin/client/fromfile/DatesFromFileTest.java:37]: I need to cherry-pick this for the 7.48.x bundle

File: drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/main/java/org/drools/workbench/services/verifier/plugin/client/Receiver.java [hunk 5]
```diff
@@ -165,6 +165,9 @@ public void sendStatus(final Status status) {
         } catch (final BuildException e) {
             poster.post(new WebWorkerException("Initialization failed: " +
                                                        e.getMessage()));
+        } catch (Exception e) {
+            poster.post(new WebWorkerException("Initialization failed with unmanaged error: " +
+                                                       e.getMessage()));
         }
```
Comment (jomarko) [inline drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/main/java/org/drools/workbench/services/verifier/plugin/client/Receiver.java:171]: What is the difference between
```java
        } catch (final BuildException e) {
            poster.post(new WebWorkerException("Initialization failed: " +
                                                       e.getMessage()));
        } catch (Exception e) {
            poster.post(new WebWorkerException("Initialization failed with unmanaged error: " +
                                                       e.getMessage()));
        }
```

and
```java
        } catch (Exception e) {
            poster.post(new WebWorkerException("Initialization failed with unmanaged error: " +
                                                       e.getMessage()));
        }
```

**Review by jomarko — COMMENTED** [review]
One question to code - try catch blocks

**Review by jomarko — APPROVED** [review]
2/2 Approving. Thank you @Rikkola

**Comment by Rikkola** [comment]
@kiegroup/gatekeepers Can I get a merge please.

**Pull request merged**